packages/pg-pool/index.js

```diff
@@ -408,3 +419,5 @@ class Pool extends EventEmitter {
   }
 }
 module.exports = Pool
+module.exports.Pool = Pool
+module.exports.PoolError = PoolError
```

Comment:
Putting `Pool` on itself doesn’t feel right pre-ESM, and it’s not easy to use via pg. Would this even need to be exported with non-colliding error codes like `ERR_PG_POOL_…`?

Reply:
Do you mean: replacing `PoolError` by constants (for instance `ERR_PG_POOL_CONNECT_TIMEOUT`, `ERR_PG_POOL_UNEXPECTED_CONDITION`, ...) ?

My goal was to implement the same kind of structure than `NoticeOrError` interface from `pg-protocol`. What would be the PROs/CONs ?

Reply:
I believe what @charmander is trying to say, is that if, instead of error codes like `Z0001`, you used [NodeJS like](https://nodejs.org/api/errors.html#nodejs-error-codes) error codes, you would not need to do `instanceof PoolError` and you could simply perform a check like:

```js
catch (err) {
  if (err.code === 'ERR_PG_POOL_UNEXPECTED_STATE') throw err // something's wrong
  if (err.code === 'ERR_PG_POOL_CONNECTION_TIMEOUT') // handle overloaded database
}
```
In this case, you would not even need to export `PoolError` at all.

Note that because of the way `node_modules` are structured, checking for `instanceof` might yield false negative in case `pg-pool` ends up twice in you `node_modules` hierarchy, making this argument 100% relevant.

This also explain his [other comment](https://github.com/brianc/node-postgres/pull/2498/files#r597562408) explaining that you don't need to explicit every possible error case. A simple `ERR_PG_POOL_UNEXPECTED_STATE` (plz find a better name 😅) is probably enough.

I believe that his first comment is that this feels right in ESM:
```js
export default Pool
export { Pool, PoolError }
```
but the following is weird in common-js (pre-ESM)
```js
module.exports = Pool
module.exports.Pool = Pool
```
because it is the same as doing `Pool.Pool = Pool` which feels odd.